src/OrleansTestingHost/Utils/StorageEmulator.cs

```diff
@@ -141,6 +141,32 @@ public static bool Stop()
             }
         }
 
+        public static bool TryCheckForAzureStorage(string dataConnectionString, out string error)
```

Comment:
I'm not sure this method should be brought to the public package, since what it is doing is not directly related to `StorageEmulator` and its behavior is not well defined. It did many things, first check `dataConnectionString` empty or not, then check if the user intend to use the emulator. If not then return true. But this method didn't check if the user really have a remote azure storage service configured. So in that path, the behavior is uncertain. At last if set to use local emulator, then start the emulator. This part is related but the `StorageEmulator` api already covers that. 

I prefer not to put this method in the public API. we can put it in `TestExtensions.csproj ` which is a common dependency for our test projects.

@jdom what do you think ?

Reply:
I agree, this isn't a general utility, so it shouldn't be part of the testing nuget package that we ship.
Having said that, StorageEmulator itself is currently part of this package, and it itself needs to be removed entirely. Adding this method temporarily is not a huge concern to me, as we have a pending task to remove the whole class. Nevertheless, it might be bad to add more stuff if we plan to deprecate it.

Reply:
BTW, I agree that the TestExtensions project is where this should go (and eventually StorageEmulator too)

Reply:
I think the proper home of this behavior depends on what pattern we want to support for extension developers.  We should, ourselves, try to develop extension libraries as we expect others to.  While we're not strictly following this ATM, I am of the opinion this should be our goal.

In general, I think this behavior makes sense as part of the azure emulator test support, so it's in the correct class.  If that class is in the wrong assembly, we should address that once we decide where it belongs, rather than peppering it across multiple projects.

Reply:
Well, the distinction is that this method just uses the StorageEmulator as a fallback, but if the connection string is not with the emulator's connection string, then it will not try to start it. So in theory it is separable from the StorageEmulator class anyway.

Reply:
I'm of the opinion that azure emulator test support should be azure team's responsibility. Since it is purely relate to them, not us. It make sense that we have it internally for our internal use. But I wouldn't agree more than that. 
BTW I didn't know StorageEmulator is going to be removed. If it is going to be removed in 2.0 beta, then sure let's merge this PR and remove it later. If not, we shouldn't add it in our public package and had it released in 2.0 beta